File: src/main/java/de/pflugradts/passbird/domain/model/password/PasswordRequirements.java
package de.pflugradts.passbird.domain.model.password;

public class PasswordRequirements {
    boolean includeSpecialCharacters;
    int passwordLength;
    private PasswordRequirements(boolean includeSpecialCharacters, int passwordLength) {
        this.includeSpecialCharacters = includeSpecialCharacters;
        this.passwordLength = passwordLength;
    }
    public static PasswordRequirements of(boolean includeSpecialCharacters, int passwordLength) {
        return new PasswordRequirements(includeSpecialCharacters, passwordLength);
    }

    public boolean isIncludeSpecialCharacters() {
        return includeSpecialCharacters;
    }

    public int getPasswordLength() {
        return passwordLength;
    }
}

File: src/main/kotlin/de/pflugradts/passbird/domain/model/password/PasswordRequirements.kt
package de.pflugradts.passbird.domain.model.password

class PasswordRequirements private constructor(val includeSpecialCharacters: Boolean, val passwordLength: Int) {
    companion object {
        fun passwordRequirementsOf(includeSpecialCharacters: Boolean, passwordLength: Int) =
            PasswordRequirements(includeSpecialCharacters, passwordLength)
    }
}
